File: spring-boot-security-jwt-auth-mongodb-master/src/main/java/com/knf/dev/security/services/EmployeeDetailsServiceImpl.java
package com.knf.dev.security.services;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import com.knf.dev.models.Employee;
import com.knf.dev.repository.EmployeeRepository;

@Service
public class EmployeeDetailsServiceImpl implements UserDetailsService {

	@Autowired
	EmployeeRepository employeeRepository;

	@Override
	@Transactional
	public UserDetails loadUserByUsername(String employeename) throws UsernameNotFoundException {

		Employee employee = employeeRepository.findByEmployeename(employeename)
				.orElseThrow(() -> new UsernameNotFoundException("Employee Not Found with username: " + employeename));

		return EmployeeDetailsImpl.build(employee);
	}

}

File: kotlin-spring-main/kotlin-springsecurity-jwt/src/main/kotlin/com/knf/dev/demo/kotlinspringsecurityjwt/security/services/EmployeeDetailsServiceImpl.kt
package com.knf.dev.demo.kotlinspringsecurityjwt.security.services

import com.knf.dev.demo.kotlinspringsecurityjwt.model.Employee
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.transaction.annotation.Transactional
import kotlin.Throws
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.stereotype.Service
import com.knf.dev.demo.kotlinspringsecurityjwt.repository.EmployeeRepository

@Service
class EmployeeDetailsServiceImpl : UserDetailsService {
    @Autowired
    var employeeRepository: EmployeeRepository? = null
    @Transactional
    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(employeename: String): UserDetails {
        val employee: Employee = employeeRepository!!.findByEmployeename(employeename)
            ?.orElseThrow { UsernameNotFoundException("Employee Not Found with username: " +
                    "$employeename") }!!
        return EmployeeDetailsImpl.build(employee)
    }
}